src/Stratis.Features.FederatedPeg/CollateralChecker.cs

```diff
@@ -182,6 +186,12 @@ private async Task<bool> UpdateCollateralInfoAsync(CancellationToken cancellatio
 
         public bool CheckCollateral(IFederationMember federationMember)
         {
+            if (!this.isInitialized)
+            {
+                this.logger.LogTrace("(-)[NOT_INITIALIZED]");
+                throw new Exception("Component is not initialized!");
```

Comment:
Yeah I'm also not sure whether or not we need to throw here? Why can't we just log and return?

Reply:
Because it's prohibited usage. It shouldn't be possible that any component calls CH before it's initialized